main.c

```diff
@@ -43,87 +52,97 @@ int main(int argc, char **argv)
   // Master communicator
   if (rank == 0) {
     // Allocates all the sets into a 2D array
-    // [[Set 1: N * N], [Set 2: N * N], ..., [Set NUM_SETS: N * N]]
-    int(*sets)[N * N] = malloc(sizeof(int[NUM_SETS][N * N]));
+    // [[Set 1: N], [Set 2: N], ..., [Set NUM_BOARDS: N]]
+    int(*sets)[N] = malloc(sizeof(int[NUM_BOARDS][N]));
 
     // Sets the initial value of the sets
-    for (int p = 0; p < NUM_SETS; p++)
+    for (int p = 0; p < NUM_BOARDS; p++)
     {
-      for (int i = 0; i < N * N; ++i)
+      for (int i = 0; i < N; ++i)
       {
         sets[p][i] = rand() % N;
       }
     }
 
     // Allocate conflict score array
-    int *conflict_scores = (int *)malloc(NUM_SETS * sizeof(int));
+    int *conflict_scores = (int *)malloc(NUM_BOARDS * sizeof(int));
 
     // Populates conflict score array
-    for (int i = 0; i < NUM_SETS; i++)
+    for (int i = 0; i < NUM_BOARDS; i++)
     {
       conflict_scores[i] = computeConflictScore(N, sets[i]);
     }
 
+    int loopNum = 0;
     // While there are still conflicts, keep trying to get new boards
-    while (sum(NUM_SETS, conflict_scores) > 0) {
+    while (sum(NUM_BOARDS, conflict_scores) > 0) {
+      printf("RUNNING LOOP: %d\n", loopNum);
+      loopNum++;
+
       // Variables for storing the new set and accompanying conflict score from slaves
-      int* optimal_set = (int*)malloc(N * N * sizeof(int));
+      int* optimal_set = (int*)malloc(N * sizeof(int));
       int newMinConf;
 
       // Communicate with all the slaves
+      // currently only set up for 1?
```

Comment:
This is set to 1 because we exclude rank 0 from doing any of the mutation stuff. We could maybe change it so rank 0 does that too but I'm not sure if that would cause any problems?

Reply:
this was fine, you were right.  i forgot to remove it before pushing the commit